lib/Resque/Job.php

```diff
@@ -188,18 +191,49 @@ public function getInstance()
 	 */
 	public function perform()
 	{
+
+		$method = 'perform';
+		$beforePerformMethod = 'setUp';
+		$tearDownMethod = 'tearDown';
 		$instance = $this->getInstance();
+		if (is_array($instance))
+		{
+			if (isset($instance[4]))
```

Comment:
So, you're saying that if `Resque_Job_Creator` returns an array, then it'll override some settings?

I'm not sure if this is the way you want to go about things... I mean - why not just have something like

```php
public function setUp($args) 
{
    return $this->realFunction($args);
}
```

I'm really not a fan of this, and think it goes significantly against the original design of resque.